plugins/tickets.py

```diff
@@ -0,0 +1,2118 @@
+                "Note prompt cancelled, no note was created."
+            )
+
+    if note:
+        # Create the note ticket
+        ticket = await create_ticket(
+            type=TicketType.NOTE,
+            modid=msg.author.id,
+            targetid=targetid,
+            created_at=dt.datetime.utcnow(),
+            created_by=msg.author.id,
+            stage=TicketStage.COMMENTED,
+            status=TicketStatus.IN_EFFECT,
+            comment=note
+        )
+
+        # Ack note creation
+        await msg.channel.send(
+            embed=discord.Embed(
+                description="[#{}]({}): Note created!".format(ticket.id,
+                                                              ticket.jump_link)
+            )
+        )
+
+
+@commands.command("tickets")
+@commands.command("ticket")
+@priv.priv("mod")
+async def cmd_ticket(msg: discord.Message, args):
+    user = msg.author
+    reply = msg.channel.send
+    no_mentions = discord.AllowedMentions.none()
+
+    S_Arg = commands.StringArg
+    UM_Arg = commands.UserMentionArg
+
+    cmd_arg = args.next_arg()
+    if not isinstance(cmd_arg, S_Arg):
+        return
+    cmd = cmd_arg.text.lower()
+
+    if cmd == "top":
+        """
+        Usage: ticket top
+        DM you the ticket at the top of your queue (if any).
+        Re-deliver the ticket at the top of your queue to your DMS.
+        """
+        mod = get_or_create_mod(user.id)
+        if not mod.current_ticket:
+            await reply("Your queue is empty, good job!")
+        else:
+            await mod.deliver()
+            if msg.channel.type != discord.ChannelType.private:
+                await reply(
+                    "Ticket #{} has been delivered to your DMs.".format(
+                        mod.current_ticket.id
+                    )
+                )
+    elif cmd == "queue":
+        """
+        Usage: ticket queue [modmention]
+        Show the specified moderator's (or your own) ticket queue.
+        """
+        modarg = args.next_arg()
+        if modarg is None or isinstance(modarg, UM_Arg):
+            modid = modarg.id if modarg is not None else user.id
+            embeds = None
+            if modid in _ticketmods:
+                mod = _ticketmods[modid]
+                tickets = mod.queue
+
+                embeds = summarise_tickets(
+                    *tickets,
+                    title='Queue for {}'.format(modid),
+                    fmt=(
+                        "[#{id}]({jump_link}): "
+                        "({status}) **{type}** for {targetid!m}>"
+                    )
+                )
+
+            if embeds:
+                await pager(
+                    msg.channel,
+                    [Page(embed=embed) for embed in embeds]
+                )
+            else:
+                await reply(
+                    util.discord.format("{!m} has an empty queue!", modid),
+                    allowed_mentions=no_mentions
```

Comment:
Very cool but we've already tagged them with our command. Allow mentioning by raw user id?